protocol/contracts/farm/facets/SiloFacet/DelegateFacet.sol

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: MIT
+ */
+
+pragma solidity ^0.7.6;
+pragma experimental ABIEncoderV2;
+
+import "./Silo.sol";
+import "../../ReentrancyGuard.sol";
+import "../../../libraries/Silo/LibDelegate.sol";
+
+/*
+ * @author Publius
+ * @title DelegateFacet handles delegating authority for Beanstalk functions.
+ */
+contract DelegateFacet is Silo {
+    mapping(bytes4 => bool) public selectorEnabled;
+
+    bytes4 public constant PLANT_DELEGATED_SELECTOR =
+        bytes4(keccak256(bytes("plantDelegated(address)")));
+
+    constructor() {
```

Comment:
Diamond facets can't use constructors because they are implementation contracts. Instead, we use separate init contracts that get executed 1 time during the `diamondCut` function. Instead of having a constructor, can we add an `InitFunctionApproval` contract with an `init()` function.